backend_new/src/services/application-flagged-set.service.ts

```diff
@@ -0,0 +1,951 @@
+
+    return {
+      items: afs,
+      meta: {
+        ...paginationInfo,
+        totalFlagged: totalFlagged,
+      },
+    };
+  }
+
+  /**
+    builds where clause for list function
+  */
+  buildWhere(params: AfsQueryParams): Prisma.ApplicationFlaggedSetWhereInput {
+    const filters: Prisma.ApplicationFlaggedSetWhereInput[] = [];
+
+    if (params.listingId) {
+      filters.push({
+        listingId: params.listingId,
+      });
+    }
+
+    if (params.view) {
+      if (params.view === View.pending) {
+        filters.push({
+          status: FlaggedSetStatusEnum.pending,
+        });
+      } else if (params.view === View.pendingNameAndDoB) {
+        filters.push({
+          status: FlaggedSetStatusEnum.pending,
+          rule: RuleEnum.nameAndDOB,
+        });
+      } else if (params.view === View.pendingEmail) {
+        filters.push({
+          status: FlaggedSetStatusEnum.pending,
+          rule: RuleEnum.email,
+        });
+      } else if (params.view === View.resolved) {
+        filters.push({
+          status: FlaggedSetStatusEnum.resolved,
+        });
+      }
+    }
+
+    return {
+      AND: filters,
+    };
+  }
+
+  /**
+    this will return 1 application flagged set or error
+  */
+  async findOne(afsId: string): Promise<ApplicationFlaggedSet> {
+    const rawAfs = await this.prisma.applicationFlaggedSet.findUnique({
+      where: {
+        id: afsId,
+      },
+      include: {
+        applications: {
+          include: {
+            applicant: true,
+          },
+        },
+        listings: true,
+      },
+    });
+
+    if (!rawAfs) {
+      throw new NotFoundException(
+        `applicationFlaggedSetId ${afsId} was requested but not found`,
+      );
+    }
+
+    return mapTo(ApplicationFlaggedSet, rawAfs);
+  }
+
+  /**
+    this resets the showConfirmationAlert on the application flagged set
+  */
+  async resetConfirmationAlert(afsId: string): Promise<SuccessDTO> {
+    await this.findOrThrow(afsId);
+    await this.prisma.applicationFlaggedSet.update({
+      data: {
+        showConfirmationAlert: false,
+      },
+      where: {
+        id: afsId,
+      },
+    });
+    return {
+      success: true,
+    };
+  }
+
+  /**
+    this will return meta info for a set of application flagged sets
+  */
+  async meta(params: AfsQueryParams): Promise<AfsMeta> {
+    const [
+      totalCount,
+      totalResolvedCount,
+      totalPendingCount,
+      totalNamePendingCount,
+      totalEmailPendingCount,
+    ] = await Promise.all([
+      this.metaHelper(params.listingId),
+      this.metaHelper(params.listingId, FlaggedSetStatusEnum.resolved),
+      this.metaHelper(params.listingId, FlaggedSetStatusEnum.pending),
+      this.metaHelper(
+        params.listingId,
+        FlaggedSetStatusEnum.pending,
+        RuleEnum.nameAndDOB,
+      ),
+      this.metaHelper(
+        params.listingId,
+        FlaggedSetStatusEnum.pending,
+        RuleEnum.email,
+      ),
+    ]);
```

Comment:
2 less queries! updated 

I think the backend should still be responsible for the calculation so I kept it as part of the dto, we just no do the calculation on the fly instead of reading from the db again!

Reply:
Thanks, @YazeedLoonat . Agreed on keeping it part of the DTO.